industrial_ci/src/tests/debians.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+
+# Copyright (c) 2021, Mathias Lüdtke
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Based on https://github.com/ros-industrial/industrial_ci/issues/697#issuecomment-876293987
+
+
+function make_repo() {
+    local repo=$1; shift
+    mkdir -p "$repo" && touch "$repo/Packages"
+}
+
+function update_repo() (
+    local repo=$1; shift
+    cd "$repo" && dpkg-scanpackages .  > Packages
+)
+
+function use_repo() (
+    local repo=$1; shift
+    ici_asroot rm -f /etc/apt/apt.conf.d/docker-clean
+    echo "deb [trusted=yes] file://$repo ./" | ici_asroot tee /etc/apt/sources.list.d/ici_debians.list  > /dev/null
+    ici_asroot apt-get -o APT::Sandbox::User=root update -qq
+    # TODO: pin this repo
+)
+
+function forward_mounts() {
+    ici_forward_mount WORKSPACE rw
+    ici_forward_mount "$WORKSPACE/archives" rw /var/cache/apt/archives
+    if [ -n "${DOCKER_PORT:-}" ]; then
+        ici_forward_variable DOCKER_HOST "$DOCKER_PORT"
+    elif [ -e /var/run/docker.sock ]; then
+        ici_forward_mount /var/run/docker.sock rw
+    fi
+}
+
+function prepare_debians() {
+    export _DEFAULT_DEBS="gnupg2 dirmngr"
+    export WORKSPACE; WORKSPACE=$(mktemp -d)
+    mkdir "$WORKSPACE/archives"
+    forward_mounts
+}
+
+function build_debian() (
+    set -e
+    local pkg_path=$1; shift
+    local repo=$1; shift
+    use_repo "$repo"
+
+    cd "$pkg_path"
+    ici_asroot apt-get build-dep -y -qq .
+    ici_exec dpkg-buildpackage -b -uc -us
+    mv ../*.deb "$repo"
+)
+
+function build_package() (
+    set -e
+    local pkg_path=$1; shift
+    local repo=$1; shift
+    export _FOLDING_TYPE=none
+    ici_source_component _FOLDING_TYPE folding
+
+    use_repo "$repo"
+    ici_quiet ici_install_pkgs_for_command dpkg-buildpackage dpkg-dev
+
+    cd "$pkg_path"
+    ici_exec bloom-generate rosdebian --ros-distro="$ROS_DISTRO" --debian-inc="ici~"
+    # https://github.com/ros-infrastructure/bloom/pull/643
+    echo 11 > debian/compat
+
+    ici_isolate debians build_debian "$current/${pkg[1]}" "$repo"
+    update_repo "$repo"
+)
+
+function run_debians() {
+    # shellcheck source=industrial_ci/src//isolation/docker.sh
+    source "${ICI_SRC_PATH}/isolation/docker.sh"
+    forward_mounts
+
+    export BUILDER=colcon
+    ici_source_builder
+    ici_run "${BUILDER}_setup" ici_quiet builder_setup
+    ici_run "setup_bloom" ici_quiet ici_install_pkgs_for_command bloom-generate python3-bloom debhelper
```

Comment:
why do you need debhelper here?

Reply:
It is not needed anymore, I will remove it!